packages/odyssey-react/src/components/Box/Box.tsx

```diff
@@ -11,128 +11,501 @@
+      }
+    }
+
+    if (Array.isArray(padding)) {
+      switch (padding.length) {
+        case 2:
+          paddingTop = padding[0];
+          paddingBottom = padding[0];
+          paddingLeft = padding[1];
+          paddingRight = padding[1];
+          break;
+        case 3:
+          paddingTop = padding[0];
+          paddingRight = padding[1];
+          paddingLeft = padding[1];
+          paddingBottom = padding[2];
+          break;
+        case 4:
+          paddingTop = padding[0];
+          paddingRight = padding[1];
+          paddingBottom = padding[2];
+          paddingLeft = padding[3];
+          break;
+        default:
+          padding = padding[0];
+      }
+    }
+
+    if (Array.isArray(borderRadius)) {
+      switch (borderRadius.length) {
+        case 2:
+          borderTopLeftRadius = borderRadius[0];
+          borderBottomRightRadius = borderRadius[0];
+          borderTopRightRadius = borderRadius[1];
+          borderBottomLeftRadius = borderRadius[1];
+          break;
+        case 3:
+          borderTopLeftRadius = borderRadius[0];
+          borderTopRightRadius = borderRadius[1];
+          borderBottomLeftRadius = borderRadius[1];
+          borderBottomRightRadius = borderRadius[2];
+          break;
+        case 4:
+          borderTopLeftRadius = borderRadius[0];
+          borderTopRightRadius = borderRadius[1];
+          borderBottomRightRadius = borderRadius[2];
+          borderBottomLeftRadius = borderRadius[3];
+          break;
+        default:
+          borderRadius = borderRadius[0];
+      }
+    }
+
+    if (Array.isArray(overflow)) {
+      switch (overflow.length) {
+        case 2:
+          overflowX = overflow[0];
+          overflowY = overflow[1];
+          break;
+        default:
+          overflowX = overflow[0];
+      }
+    }
 
-  const componentClass = useCx(
-    styles.root,
-    borderColor !== "none" && styles.borderBase,
-    borderColor && styles[`borderColor${convertProp(borderColor)}`],
-    hoverBorderColor !== "none" && styles.borderBase,
-    hoverBorderColor &&
-      styles[`hoverBorderColor${convertProp(hoverBorderColor)}`],
-    padding && styles[`padding${convertProp(padding)}`],
-    borderRadius && styles[`borderRadius${convertProp(borderRadius)}`],
-    boxShadow && styles[`boxShadow${convertProp(boxShadow)}`],
-    hoverBoxShadow && styles[`hoverBoxShadow${convertProp(hoverBoxShadow)}`],
-    focusRing && styles[`focusRing${convertProp(focusRing)}`],
-    display && styles[`display${convertProp(display)}`],
-    flexGrow && styles[`flexGrow${convertProp(flexGrow)}`],
-    marginRight && styles[`marginRight${convertProp(marginRight)}`],
-    alignItems && styles[`alignItems${convertProp(alignItems)}`],
-    marginBottom && styles[`marginBottom${convertProp(marginBottom)}`],
-    cursor && styles[`cursor${convertProp(cursor)}`],
-    userSelect && styles[`userSelect${convertProp(userSelect)}`]
-  );
+    const componentClass = useCx(
```

Comment:
this should also accept arbitrary classes as part of the API yes? We need to allow `className` to be passed into `Box` so that it doesn't have to contain every style for all use cases.